src/App.jsx

```diff
@@ -1,33 +1,53 @@
+import { useState } from 'react';
+
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-export const App = () => (
-  <div className="page">
-    <div className="page-content">
-      <div className="box">
-        <div className="field">
-          {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
-          <label htmlFor="search-query" className="label">
-            Search movie
-          </label>
-
-          <div className="control">
-            <input
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+export const App = () => {
+  const [condition, setCondition] = useState('');
+  const visibleMovies = getvisibleMovies(moviesFromServer, condition);
+
+  function getvisibleMovies(movies, query) {
+    const filteredMovies = [...movies];
+
+    if (query) {
+      return filteredMovies.filter(movie => movie.title
+        .toLowerCase().includes(query.toLowerCase().trim()) || movie.description
+        .toLowerCase().includes(query.toLowerCase().trim()));
+    }
+
+    return filteredMovies;
+  }
+
+  return (
+    <div className="page">
+      <div className="page-content">
+        <div className="box">
+          <div className="field">
+            {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+            <label htmlFor="search-query" className="label">
+              Search movie
+            </label>
+
+            <div className="control">
+              <input
+                type="text"
+                id="search-query"
+                className="input"
+                placeholder="Type search word"
+                onChange={event => setCondition(event.target.value)}
```

Comment:
make handler and pass reference here
```
const handleChangeQuery = (event: React.ChangeEvent<HtmlInputElement>) => setCondition(event.target.value)
```
```suggestion
                onChange={handleChangeQuery}
```